packages/core/src/RenderingEngine/BaseVolumeViewport.ts

```diff
@@ -717,54 +718,86 @@ abstract class BaseVolumeViewport extends Viewport {
+      // Any view is compatible with a point position
+      return;
+    }
+    const { viewPlaneNormal } = this.getCamera();
+    if (
+      isCompatible(viewPlaneNormal, inPlaneVector2) &&
+      isCompatible(viewPlaneNormal, inPlaneVector1)
+    ) {
+      // Orthogonal view to the current view, so no change.
+      return;
     }
 
-    // const planeUp = <Point3>vec3.normalize(vec3.create(), inPlaneVector1);
-    if (inPlaneVector2) {
-      const planeNormal = <Point3>(
-        vec3.cross(vec3.create(), inPlaneVector1, inPlaneVector2)
-      );
-      vec3.normalize(planeNormal, planeNormal);
-      return this.setViewReference({
-        FrameOfReferenceUID,
-        cameraFocalPoint: point,
-        viewPlaneNormal: planeNormal,
-        // viewUp: planeUp,
-      });
+    const acquisition = this._getAcquisitionPlaneOrientation();
+    if (
+      isCompatible(acquisition.viewPlaneNormal, inPlaneVector2) &&
+      isCompatible(acquisition.viewPlaneNormal, inPlaneVector1)
+    ) {
+      // Orthogonal view to the current view, so no change.
+      this.setCamera(acquisition);
+      return;
     }
-    const dotNormal = vec3.dot(viewPlaneNormal, inPlaneVector1);
-    if (isEqual(dotNormal, 0)) {
-      return this.setViewReference({
-        FrameOfReferenceUID,
-        viewPlaneNormal,
-        cameraFocalPoint: point,
-      });
+    for (const orientation of <{ viewPlaneNormal: Point3 }[]>(
+      Object.values(mprCameraValues)
+    )) {
+      if (
+        isCompatible(orientation.viewPlaneNormal, inPlaneVector2) &&
+        isCompatible(orientation.viewPlaneNormal, inPlaneVector1)
+      ) {
+        // Orthogonal view to the current view, so no change.
+        this.setCamera(orientation);
+        return;
+      }
     }
 
     const planeNormal = <Point3>(
-      vec3.cross(vec3.create(), viewPlaneNormal, inPlaneVector1)
+      vec3.cross(
+        vec3.create(),
+        inPlaneVector1,
+        inPlaneVector2 || acquisition.viewPlaneNormal
+      )
     );
     vec3.normalize(planeNormal, planeNormal);
-    return this.setViewReference({
-      FrameOfReferenceUID,
-      viewPlaneNormal: planeNormal,
-      cameraFocalPoint: point,
-      // viewUp: planeUp,
+    this.setCamera({ viewPlaneNormal: planeNormal });
+  }
+
+  /**
+   * Sets the view reference given a referenced plane and the current
+   * view plane normal being applied.
+   * This will use the existing normal if compatible, otherwise will calculate
+   * a new view plane normal as the referenced plane normal, or else the
+   * cross product of the existing view plane normal and the inPlaneVector1
+   */
+  public setViewPlane(planeRestriction: PlaneRestriction) {
+    const { point, inPlaneVector1, inPlaneVector2 } = planeRestriction;
+
+    this.setBestOrentation(inPlaneVector1, inPlaneVector2);
+
+    const { viewPlaneNormal, focalPoint, position } = this.getCamera();
+
+    const delta = vec3.subtract([0, 0, 0], focalPoint, point);
```

Comment:
I think something in this logic is wrong somehow at least some of the time as the new focal point/position seems to get out of view somehow.